crates/metadata/lib.rs

```diff
@@ -267,10 +267,15 @@ impl Metadata {
             cargo_args.push("--no-default-features".into());
         }
 
-        let mut all_rustdoc_args = self.rustdoc_args.clone();
+        // Unconditionnaly set `--cfg docsrs` as it has become a de-facto way to
+        // distinguish docs.rs.
+        //
+        // See https://github.com/rust-lang/docs.rs/issues/2389.
+        let mut all_rustdoc_args = vec!["--cfg".into(), "docsrs".into()];
+        all_rustdoc_args.extend_from_slice(&self.rustdoc_args);
         all_rustdoc_args.extend_from_slice(rustdoc_args);
 
-        if !self.rustc_args.is_empty() || !all_rustdoc_args.is_empty() {
+        if !self.rustc_args.is_empty() || all_rustdoc_args.len() > 2 {
```

Comment:
I think we might be able to get rid of this now, it was from when `--config` was unstable, we're not adding any other unstable flags below other than `-Z` ones which shouldn't need `-Zunstable-options`.

Reply:
Done.